pkg/securitycontext/admission/admission.go

```diff
@@ -0,0 +1,284 @@
+
+			// Validate the copy of the container against the security context provider
+			container.SecurityContext = context
+			errs := provider.ValidateSecurityContext(pod, &container)
+			if len(errs) != 0 {
+				continue
+			}
+
+			// Set the security context on actual container
+			pod.Spec.Containers[i].SecurityContext = context
+			if pod.Annotations == nil {
+				pod.Annotations = make(map[string]string, 0)
+			}
+			pod.Annotations[validatedSCCAnnotation] = providerName
+			continue Containers
+		}
+
+		// If we have reached this code, we couldn't find an SCC that matched
+		// the requested security context for this container
+		//
+		// TODO: better error message
+		return errors.NewBadRequest("Unable to find a valid security context constraint for container")
+	}
+
+	return nil
+}
+
+// getMatchingSecurityContextConstraints returns constraints from the store that match the group,
+// uid, or user of the service account.
+func (c *constraint) getMatchingSecurityContextConstraints(ns string, serviceAccount *kapi.ServiceAccount) ([]*kapi.SecurityContextConstraints, error) {
+	userInfo := serviceaccount.UserInfo(ns, serviceAccount.Name, string(serviceAccount.UID))
+	matchedConstraints := make([]*kapi.SecurityContextConstraints, 0)
+
+	for _, c := range c.store.List() {
+		constraint, ok := c.(*kapi.SecurityContextConstraints)
+		if !ok {
+			return nil, errors.NewInternalError(fmt.Errorf("Error converting object from store to a security context constraint: %v", c))
+		}
+		for _, userGroup := range userInfo.GetGroups() {
+			if constraintSupportsGroup(userGroup, constraint.Groups) {
+				matchedConstraints = append(matchedConstraints, constraint)
+				break
+			}
+		}
+
+		for _, user := range constraint.Users {
+			if userInfo.GetName() == user || userInfo.GetUID() == user {
+				matchedConstraints = append(matchedConstraints, constraint)
+				break
+			}
+		}
+	}
+
+	return matchedConstraints, nil
+}
+
+// getPreallocatedUIDRange retrieves the annotated value from the service account, splits it to make
+// the min/max and formats the data into the necessary types for the strategy options.
+func getPreallocatedUIDRange(ns *kapi.Namespace) (min *int64, max *int64, err error) {
+	annotationVal, ok := ns.Annotations[allocator.UIDRangeAnnotation]
+	if !ok {
+		err = errors.NewInternalError(fmt.Errorf("Unable to find annotation %s", allocator.UIDRangeAnnotation))
+		return
+	}
+	annotationSlice := strings.Split(annotationVal, "/")
+	if len(annotationSlice) != 2 {
+		err = errors.NewInternalError(fmt.Errorf("Invalid annotation value %s.  Expected format {max}/{min}", annotationVal))
+		return
+	}
+
+	iMin, e := strconv.ParseInt(annotationSlice[1], 10, 64)
+	if e != nil {
+		err = e
+		return
+	}
+	min = &iMin
+
+	iMax, e := strconv.ParseInt(annotationSlice[0], 10, 64)
+	if e != nil {
+		err = e
+		return
+	}
+	max = &iMax
+	return
+}
+
+// getPreallocatedLevel gets the annotated value from the service account and ensures that
+// s0 is added.
```

Comment:
reviewing the label allocator code again I think this may be altogether wrong.  Will update.